test/emails/post-emails.api-test.ts

```diff
@@ -0,0 +1,220 @@
+    makeRequestWithoutAuth: (incorrectAuth?: IncorrectAuthArg) => api.postWithoutAuth(mdmPath, request, incorrectAuth?.headerName, incorrectAuth?.headerValue),
+  });
+
+  it('returns a 201 response with receipt for sent email', async () => {
+    const { status, body } = await api.post(mdmPath, request, { govUkNotifyKey });
+
+    expect(status).toBe(201);
+    expect(body).toStrictEqual(postEmailsResponse[0][0]);
+  });
+
+  it('calls NotifyClient.sendEmail', async () => {
+    await api.post(mdmPath, request, { govUkNotifyKey });
+
+    expect(sendEmailMethodMock).toHaveBeenCalledWith(
+      request[0].templateId,
+      request[0].sendToEmailAddress,
+      expect.objectContaining({
+        personalisation: request[0].personalisation,
+        // Check if autogenerated `reference` matches expected pattern.
+        // eslint-disable-next-line security/detect-non-literal-regexp
+        reference: expect.stringMatching(new RegExp(`^${request[0].templateId}-\\d*$`)),
+      }),
+    );
+  });
+
+  it('calls NotifyClient.sendEmail with custom reference', async () => {
+    const customReference = valueGenerator.string({ length: 10 });
+
+    await api.post(mdmPath, [{ ...request[0], reference: customReference }], { govUkNotifyKey });
+
+    expect(sendEmailMethodMock).toHaveBeenCalledWith(request[0].templateId, request[0].sendToEmailAddress, {
+      personalisation: request[0].personalisation,
+      reference: customReference,
+    });
+  });
+
+  it('returns a 400 response if header `govUkNotifyKey` is missing', async () => {
+    const { status, body } = await api.post(mdmPath, request);
+
+    expect(status).toBe(400);
+    expect(body).toStrictEqual({ error: 'Bad Request', message: 'Header "govUkNotifyKey" is required', statusCode: 400 });
+  });
+
+  it.each([
+    {
+      error: 'Bad Request',
+      expectedStatus: 400,
+    },
+    {
+      error: 'Unauthorized',
+      expectedStatus: 401,
+    },
+    {
+      error: 'Forbidden',
+      expectedStatus: 403,
+    },
+  ])('returns a $expectedStatus response if notify client response with status $expectedStatus', async ({ error, expectedStatus }) => {
+    jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.reject(generateNotifyError(expectedStatus, errorMessage)));
+
+    const { status, body } = await api.post(mdmPath, request, { govUkNotifyKey });
+
+    expect(status).toBe(expectedStatus);
+    expect(body).toMatchObject({
+      error,
+      message: expect.arrayContaining([errorMessage]),
+      statusCode: expectedStatus,
+    });
+  });
+
+  it('returns a 500 response if notify client response with status 500', async () => {
+    jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.reject(generateNotifyError(500)));
+
+    const { status, body } = await api.post(mdmPath, request, { govUkNotifyKey });
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({
+      statusCode: 500,
+      message: 'Internal server error',
+    });
+  });
```

Comment:
Could this be part of the `it.each` block above it to make the code more DRY?

Reply:
Thanks, moving 500 test to `it.each` worked, and produced different, better, body.